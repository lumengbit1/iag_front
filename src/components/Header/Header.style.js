import styled from 'styled-components';

export const Root = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  background-color: white;
  border-bottom: solid 1px black;
`;

export const Title = styled.div`
  font-size: 28px;
  font-weight: bold;
`;

export const Container = styled.div`
  display: flex;
  align-items: center;
  margin: 20px 0;
`;

export const Hint = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 300px;
  height: 50px;
  border: dotted 1px black;
  box-sizing: border-box;

  font-size: 24px;
  margin-left: 20px;
`;
