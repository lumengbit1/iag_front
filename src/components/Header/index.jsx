import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import {
  Root,
  Container,
  Hint,
  Title,
} from './Header.style';

const Header = (props) => (
  <Root>
    <Title>
      Guess The Password
    </Title>
    <Container>
      Hint --------&gt;
      <Hint>
        {props.hintValue}
      </Hint>
    </Container>
  </Root>
);

Header.propTypes = {
  hintValue: ImmutablePropTypes.list.isRequired,
};

export default Header;
