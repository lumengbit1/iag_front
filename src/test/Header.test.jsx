import React from 'react';
import 'jest-styled-components';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import { fromJS } from 'immutable';
import Header from '../components/Header';
import getReducer from '../reducers/get';
import postReducer from '../reducers/post';

const store = createStore(combineReducers({ getReducer, postReducer }), applyMiddleware(thunk));

const mockData = fromJS([
  {
    correct: false,
    highlight: [],
    hint: 'test',
    answer: '1111',
  },
  {
    correct: false,
    highlight: [1],
    hint: 'test',
    answer: '2222',
  },
]);

const mockhint = fromJS([
  {
    correct: false,
    highlight: [],
    hint: 'test',
    answer: '1111',
  },
]);

describe('Render Test', () => {
  it('case: expect rendering correct', () => {
    const { container } = render(
      <Provider store={store}>
        <Header reponseValue={mockData} hintValue={mockhint} />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
