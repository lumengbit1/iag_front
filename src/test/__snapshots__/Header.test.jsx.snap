// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test case: expect rendering correct 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  background-color: white;
  border-bottom: solid 1px black;
}

.c1 {
  font-size: 28px;
  font-weight: bold;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 20px 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
  height: 50px;
  border: dotted 1px black;
  box-sizing: border-box;
  font-size: 24px;
  margin-left: 20px;
}

@media screen and (max-width:768px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (max-width:768px) {
  .c3 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Guess The Password
    </div>
    <div
      class="c2"
    >
      Hint
      <div
        class="c3"
      >
        --------&gt;
      </div>
      <div
        class="c4"
      >
        correct
        highlight
        hint
        test
        answer
        1111
      </div>
    </div>
  </div>
</div>
`;
